ruby-tools/src/main/java/de/saumya/mojo/ruby/gems/GemsInstaller.java

```diff
@@ -252,6 +258,14 @@ private Script maybeAddArtifact(Script script, final Artifact artifact)
         return script;
     }
 
+    private JRubyVersion getVersion() {
+        try {
+            return this.factory.getVersion();
+        } catch (Exception e) {
+            return null;
```

Comment:
do we need really need to swallow silently?

Reply:
I would argue a not aborting with hard failure here is not the end of the world...but...

 1. It should warn that it cannot detect version (I think what @kares wants?)
 1. It should not emit no rdoci/ri since it really does not know what jruby version it is (and therefore will work for all versions of jruby).

Reply:
You are right @enebo, in some way the idea was to keep it as similar as possible with current behauviour. So if for some reason the version could not be extracted*, I prefered a more "optimistic" approach and fall back to current behauviour. The code is 1.5 so could not use Optional which may had been nicer.
About the warning is not so easy, I can inject the Logger from the mojo when called from AbstractGemMojo, but not when called from RailsService. To avoid extra checks I could use Slf4j if it's ok, Maven picks it up. or just create a Slf4j if none is injected.

*Note we would need to parse several differents formats for differents Ruby implementions and for now I focused only on jruby.

Reply:
@abelsromero Sounds reasonable to me.

Reply:
After a second thought I think it doesn't make sense to treat that in GemInstaller. I move it to ScriptFactory and added a WARN message there.
wdyt?